packages/victory-core/src/victory-util/data.js

```diff
@@ -91,6 +92,64 @@ function cleanData(dataset, props) {
   });
 }
 
+// This method will remove data points that fall outside of the desired domain (non-continuous charts only)
+function formatDataFromDomain(dataset, props) {
+  // domain for checking data points that are outside of the bounds
+  // symbol to determine chart types with a single y value point
+  // interpolation for determining chart types with continuous data points that shouldn't be altered
+  // errorX/Y for determining chart types with error whiskers that shouldn't be altered here
+  const { domain, symbol, interpolation, errorX, errorY } = props;
+
+  if (interpolation || errorX || errorY) return dataset;
+
+  const minDomainX =
+    !domain || !domain.x ? Domain.getMinFromProps(props, "x") : Collection.getMinValue(domain.x);
+  const maxDomainX =
+    !domain || !domain.x ? Domain.getMaxFromProps(props, "x") : Collection.getMaxValue(domain.x);
+  const minDomainY =
+    !domain || !domain.y ? Domain.getMinFromProps(props, "y") : Collection.getMinValue(domain.y);
+  const maxDomainY =
+    !domain || !domain.y ? Domain.getMaxFromProps(props, "y") : Collection.getMaxValue(domain.y);
+
+  const exists = (val) => val !== undefined;
+
+  // eslint-disable-next-line complexity
+  return dataset.map((datum) => {
+    let { _x, _y, _y0, _y1 } = datum;
+
+    // don't alter data points if _y is an array of values
+    if (Array.isArray(_y)) return datum;
+
+    // single x point less than min domain
+    if (exists(_x) && _x < minDomainX) _x = null;
+    // single x point greater than max domain
+    if (exists(_x) && _x > maxDomainX) _x = null;
+
+    // single y point less than min domain
+    if (exists(_y) && !exists(_y0) && _y < minDomainY) {
+      if (exists(symbol)) _y = null;
+      else _y = minDomainY;
+    }
+    // single y point greater than max domain
+    if (exists(_y) && !exists(_y0) && _y > maxDomainY) {
+      if (exists(symbol)) _y = null;
+      else _y = maxDomainY;
```

Comment:
I don't think this is the correct behavior. I think it could lead to misleading charts. I think we only want to clamp the baselines. I think unfortunately the correct thing to do is to let the value escape the domain when it is outside.  (Where baseline = `_y0` and value = `_y` or `_y1`)

Here are the situations as I see them for charts that have baselines, and what I think ought to happen from a correctness standpoint. Let me know what you think.

1) _both_ the baseline and the value are below the min or both are above the max -> null the whole point
2) the baseline is below the min, and the value is above the min -> clamp the baseline to the min and leave the value alone
3) the baseline is above the max, and the value is below the max -> clamp the baseline to the max and leave the value alone. 

Yes, this will allow some values to escape the desired domain, but users will still have the option of using clipped groups if they want to hide that. I think it does reduce the inadvertant display of inaccurate data and potential incorrect stacking.

Reply:
I agree with this and made the updates you suggest.  It actually makes it a little safer and closer to the current functionality which I like.  Definitely want to make less unexpected data point edits.